pkg/tsdb/influxdb/response_parser_test.go

```diff
@@ -1,201 +1,337 @@
 package influxdb
 
 import (
-	"encoding/json"
-	"fmt"
+	"io/ioutil"
+	"strings"
 	"testing"
 
-	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/stretchr/testify/require"
 )
 
+func responseFromJSON(t *testing.T, text string) *Response {
+	buf := ioutil.NopCloser(strings.NewReader(text))
+	response, err := parseJSON(buf)
+	require.NoError(t, err)
+	return response
+}
+
+func decodedFrames(t *testing.T, result *plugins.DataQueryResult) data.Frames {
+	decoded, err := result.Dataframes.Decoded()
+	require.NoError(t, err)
+	return decoded
+}
+
 func TestInfluxdbResponseParser(t *testing.T) {
 	t.Run("Influxdb response parser should parse everything normally", func(t *testing.T) {
 		parser := &ResponseParser{}
 
-		cfg := setting.NewCfg()
-		err := cfg.Load(&setting.CommandLineArgs{
-			HomePath: "../../../",
-		})
-		require.NoError(t, err)
+		response := responseFromJSON(t, `
```

Comment:
Not exactly against it, but I think is a bit cleaner to construct the object instead of hardcoding a JSON and using `responseFromJSON` and explicitly. 

Could be easier to change some fields on the fly inside a very test also.

Reply:
@dsotirakis hmmm.. the original code worked that way (if i understand you correctly):
https://github.com/grafana/grafana/blob/master/pkg/tsdb/influxdb/response_parser_test.go#L33

the reason i changed it is that there is a bug in that test i linked above.
BAD:    `{json.Number("111"), json.Number("null"), json.Number("333")},`
GOOD: `{json.Number("111"), nil, json.Number("333")},`

you see, when `null` arrives in JSON, it gets converted to `nil` in the json-conversion-code, not into `json.Number("null")`.  i have only found out about this problem by running the code and printing out results.  and i was worried that there can be other bugs with this approach, so i went with raw-json. (basically, one makes an assumption about how the json-to-golang-struct conversion goes, and if that assumption is wrong, things go bad).

what do you think?

Reply:
Yeah what I meant really was instead of having static JSON files, have an object instead, like:

```
response := Response{
			Results: []Result{
				{
					Series: []Row{
						{
							Name: "cpu",
							Tags: map[string]string{
								"datacenter": "America",
							},
							Columns: []string{
								"time",
								"mean",
								"sum",
							},
							Values: [][]interface{}{
								{
									...
								},
							},
						},
					},
				},
			},
		}
```

Reply:
So then the parsing would be unnecessary

Reply:
(this was resolved based on a disussion with the json-parsing being moved into `response_parser.go`